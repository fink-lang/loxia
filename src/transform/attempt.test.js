import {fink2js} from '../testing';


test('compiles attempt', ()=> {
  expect(
    fink2js(`
      foo1 = ():
        [item, err] = attempt:
          bar = shrub()
          bar + ni()
        [item, err]

      foo2 = (): attempt: 1/0
    `)
  ).toMatchSnapshot();
});


