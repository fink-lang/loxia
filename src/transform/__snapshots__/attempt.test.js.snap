// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiles attempt 1`] = `
"const foo1 = () => {
  let _do_result;

  try {
    const bar = shrub();
    _do_result = [null, bar + ni()];
  } catch ($error1) {
    _do_result = [$error1, null];
  }

  const [item, err] = _do_result;
  _do_result = undefined;
  return [item, err];
};

const foo2 = () => try {
  return [null, 1 / 0];
} catch ($error2) {
  return [$error2, null];
};"
`;
